src/dex/swaap-v2/swaap-v2.ts

```diff
@@ -0,0 +1,713 @@
+        levels = asksAndBids.asks;
+      } else {
+        levels = this.invertPrices(asksAndBids.bids);
+      }
+    } else {
+      if (srcToken.address === asksAndBids.base!) {
+        levels = asksAndBids.bids;
+      } else {
+        levels = this.invertPrices(asksAndBids.asks);
+      }
+    }
+    return this.computeLevelsQuote(amounts, levels, srcToken, destToken, side);
+  }
+
+  inversePrice(priceLevel: SwaapV2PriceLevel): SwaapV2PriceLevel {
+    return {
+      level: priceLevel.level * priceLevel.price,
+      price: 1 / priceLevel.price,
+    };
+  }
+
+  invertPrices(levels: SwaapV2PriceLevel[]): SwaapV2PriceLevel[] {
+    return levels.map(pl => this.inversePrice(pl));
+  }
+
+  computeLevelsQuote(
+    amounts: BigNumber[],
+    levels: SwaapV2PriceLevel[],
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+  ): bigint[] {
+    const size = levels.length;
+    if (size === 0) {
+      return amounts.map(_ => BigInt(0));
+    }
+    return amounts.map((amount: BigNumber) => {
+      if (amount.isZero() || amount.lt(BigNumber(levels[0].level))) {
+        return BigInt(0);
+      }
+
+      let i = 0;
+      let output = BN_0;
+      let previousLevel = BN_0;
+      let remaininig = BigNumber(amount);
+      let enoughLiquidity = false;
+      while (i < size) {
+        const levelDelta = BigNumber(levels[i].level).minus(previousLevel);
+        if (levelDelta.lte(remaininig)) {
+          output = output.plus(levelDelta.times(levels[i].price));
+          remaininig = remaininig.minus(levelDelta);
+          previousLevel = BigNumber(levels[i].level);
+          i += 1;
+        } else {
+          output = output.plus(remaininig.times(levels[i].price));
+          enoughLiquidity = true;
+          break;
+        }
+      }
+      if (enoughLiquidity) {
+        return BigInt(
+          output
+            .multipliedBy(
+              getBigNumberPow(
+                side === SwapSide.BUY ? srcToken.decimals : destToken.decimals,
+              ),
+            )
+            .toFixed(0),
+        );
+      }
+      return BigInt(0); // not enough liquidity
+    });
+  }
+
+  async getCachedLevels(): Promise<Record<string, SwaapV2PriceLevels> | null> {
+    const cachedLevels = await this.dexHelper.cache.get(
+      this.dexKey,
+      this.network,
+      getPriceLevelsCacheKey(this.dexKey),
+    );
+
+    if (cachedLevels) {
+      return JSON.parse(cachedLevels) as Record<string, SwaapV2PriceLevels>;
+    }
+
+    return null;
+  }
+
+  normalizeToken(token: Token): Token {
+    return {
+      address: normalizeTokenAddress(token.address),
+      decimals: token.decimals,
+    };
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<SwaapV2Data>> {
+    const normalizedSrcToken = this.normalizeToken(srcToken);
```

Comment:
The whole function better to be wrapped in global try catch, to log if something wrong happened here

Reply:
@Verisana try-catch is added https://github.com/paraswap/paraswap-dex-lib/pull/405/commits/f119eecc796488c63d0c0cd56f79cf6a31ddf808